File: tests/decrypt_test.rs
use aegis_cli::decrypt::decrypt_vault;

#[test]
fn test_decrypt_test_vault() {
    let vault_path = "test/resources/aegis_encrypted.json";
    let content = std::fs::read_to_string(vault_path).unwrap();
    
    // Password is "test" per the decrypt.py example
    let result = decrypt_vault(&content, "test");
    assert!(result.is_ok());
    
    let vault_content = result.unwrap();
    assert!(!vault_content.entries.is_empty());
}

File: internal/vault/decrypt_test.go
package vault_test

import (
	"testing"
	"aegis-cli/internal/vault"
)

const testVaultPath = "../../test/resources/aegis_encrypted.json"

func TestDecryptVault(t *testing.T) {
	// Password for test fixture is "test"
	result, err := vault.LoadAndDecrypt(testVaultPath, "test")
	if err != nil {
		t.Fatalf("failed to decrypt vault: %v", err)
	}

	if len(result.Content.Entries) == 0 {
		t.Error("expected at least one entry in decrypted vault")
	}
}

func TestDecryptVaultWrongPassword(t *testing.T) {
	_, err := vault.LoadAndDecrypt(testVaultPath, "wrongpassword")
	if err == nil {
		t.Fatal("expected error for wrong password, got nil")
	}
}
